lib/project_types/extension/models/specification_handlers/beacon_extension.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+require "base64"
+require "fileutils"
+require "json"
+
+module Extension
+  module Models
+    module SpecificationHandlers
+      class BeaconExtension < Default
+        SCRIPT_FILE = "beacon_extension.js"
+        CONFIG_FILE = "config.json"
```

Comment:
@lemonmade is there precedent as to what the partner-provided scripts should be? what do scripts/checkout extensions name their files?

Reply:
UI extensions use `extension.config.yml`, Scripts use `script.config.yml`.

Reply:
Hm, the naming convention might be a bit confusing.

We have two "config-like" inputs we need from partners:

1. A schema that defines the merchant configuration they will be providing us
2. A boolean indicating whether the partner code being provided should be run unsandboxed or sandboxed

Sandor perhaps we should combine both of these into a single YML file. Looking at `scripts` as an example: https://shopify-dev-staging3.shopifycloud.com/apps/scripts/configuration

Perhaps we should have something like this:

```
version: 2
sandbox: true
configuration:
  type: object
  fields:
    configurationKey:
      name: My configuration field
      description: This is a description
      type: single_line_text_field
      validations:
      - type: min
        value: "3"
```

Although I think naming the schema attribute `configuration` is a bit misleading here